CONTRIBUTING.md

```diff
@@ -1,17 +1,58 @@
 # Contributing
+To make contributions to this charm, you'll need a working Juju development setup.
 
-To make contributions to this charm, you'll need a working [Juju development setup](https://juju.is/docs/sdk/dev-setup).
+## Prerequisites
+### Charmcraft installation
+Charmcraft depends on LXD to build the charms in a container matching the target base(s).
 
-You can use the environments created by `tox` for development:
+To install Charmcraft and LXD run the following commands:
+```shell
+sudo snap install --classic charmcraft
+sudo snap install lxd
+sudo adduser $USER lxd
+newgrp lxd
+lxd init --auto
+```
+
+### MicroK8s installation
+To install MicroK8s run the following commands:
+```shell
+sudo snap install microk8s --channel=1.27-strict/stable
+sudo usermod -a -G snap_microk8s $USER
+newgrp snap_microk8s
+sudo microk8s enable hostpath-storage
+```
 
+### Install Juju
+To install Juju run the following command:
+```shell
+sudo snap install juju
+```
+To bootstrap a Juju controller on the MicroK8s instance run the following command:
```

Comment:
You can remove all the `to <xyz>`. Having the "to" in the beginning makes it unclear whether devs have to do it or not. Be prescriptive and specific. Here we can simply say:

Install Juju:
```shell
sudo snap install juju
```

Install MicroK8s:

```shell
...
```